React/Services/IReactService.cs

```diff
@@ -1,11 +1,34 @@
 ﻿using Microsoft.AspNetCore.Html;
 using Microsoft.AspNetCore.Mvc.Rendering;
 using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
 
 namespace React.Services;
 
-internal interface IReactService
+internal class ReactServiceRenderToStringAsyncExecutionTimeDecorator : IReactService
 {
-    Task<IHtmlContent> RenderToStringAsync<T>(IHtmlHelper htmlHelper, string componentName, T props);
-    IHtmlContent InitJavascript(IHtmlHelper htmlHelper);
+    private readonly IReactService _service;
+    private readonly ILogger<ReactServiceRenderToStringAsyncExecutionTimeDecorator> _logger;
+    
+    public ReactServiceRenderToStringAsyncExecutionTimeDecorator(IReactService service, ILogger<ReactServiceRenderToStringAsyncExecutionTimeDecorator> logger)
```

Comment:
please consider renaming to pattern `addedBehavior DecoratedThing Decorator` so in this case: `ExecutionTimeReactServiceRenderToStringAsyncDecorator` as this vastly increase readability